internal/connector/internal/services/vault/vault_service_aws.go

```diff
@@ -15,8 +15,10 @@ var OwnerResourceTagKey = "owner-resource"
 var _ VaultService = &awsVaultService{}
 
 type awsVaultService struct {
-	secretCache  *secretcache.Cache
-	secretClient *secretsmanager.SecretsManager
+	secretCache        *secretcache.Cache
+	secretClient       *secretsmanager.SecretsManager
+	secretPrefixEnable bool
```

Comment:
Is an enabling boolean necessary?
After all if you specify a prefix you implicitly enable the feature right? What is the point of specifying a prefix and disabling the feature?

Reply:
Yes, it's necessary if one also wants to support a default value for a prefix. Otherwise it's a nightmare to figure out whether it's enabled or not by having to look at every point of configuration i.e, code, template, and app-services config. With a switch in the app-services config, it will be self documenting.